src/StreamableFeesLocker.sol

```diff
@@ -0,0 +1,340 @@
+            lockDuration: uint32(lockDuration)
+        });
+
+        emit Lock(tokenId, beneficiaries, block.timestamp + lockDuration);
+
+        return ERC721TokenReceiver.onERC721Received.selector;
+    }
+
+    /// @notice Accrues and distributes fees for a position
+    /// @param tokenId ID of the position to accrue fees for
+    function distributeFees(
+        uint256 tokenId
+    ) external nonReentrant {
+        PositionData memory position = positions[tokenId];
+        require(position.startDate != 0, PositionNotFound());
+        require(position.isUnlocked != true, PositionAlreadyUnlocked());
+
+        // Get pool info
+        (PoolKey memory poolKey,) = positionManager.getPoolAndPositionInfo(tokenId);
+
+        // Get the amount of fees to claim
+        bytes memory actions = abi.encodePacked(uint8(Actions.DECREASE_LIQUIDITY), uint8(Actions.TAKE_PAIR));
+        bytes[] memory params = new bytes[](2);
+        params[0] = abi.encode(tokenId, 0, 0, 0, new bytes(0));
+        params[1] = abi.encode(poolKey.currency0, poolKey.currency1, address(this));
+
+        positionManager.modifyLiquidities(abi.encode(actions, params), block.timestamp);
+
+        BeneficiaryData[] memory beneficiaries = position.beneficiaries;
+
+        (uint256 currency0ToDistribute, uint256 currency1ToDistribute) = _updateCurrencyBalances(position, poolKey);
+
+        uint256 amount0Distributed;
+        uint256 amount1Distributed;
+        address beneficiary;
+        for (uint256 i; i < beneficiaries.length; ++i) {
+            beneficiary = beneficiaries[i].beneficiary;
+            uint256 shares = beneficiaries[i].shares;
+
+            // Calculate share of fees for this beneficiary
+            uint256 amount0 = currency0ToDistribute * shares / WAD;
+            uint256 amount1 = currency1ToDistribute * shares / WAD;
+
+            _distributeFees(beneficiary, poolKey, amount0, amount1);
+
+            amount0Distributed += amount0;
+            amount1Distributed += amount1;
+        }
+
+        // Distribute the remaining fees to the last beneficiary
+        uint256 amount0Remaining =
+            currency0ToDistribute > amount0Distributed ? currency0ToDistribute - amount0Distributed : 0;
+        uint256 amount1Remaining =
+            currency1ToDistribute > amount1Distributed ? currency1ToDistribute - amount1Distributed : 0;
+        _distributeFees(beneficiary, poolKey, amount0Remaining, amount1Remaining);
+
+        // Note: For no-op governance, if recipient is DEAD_ADDRESS (0xdead), the position will be permanently locked
+        // and beneficiaries can collect fees in perpetuity
+        if (block.timestamp >= position.startDate + position.lockDuration && position.recipient != DEAD_ADDRESS) {
+            position.isUnlocked = true;
+
+            // Transfer the position to the recipient
+            ERC721(address(positionManager)).safeTransferFrom(address(this), position.recipient, tokenId, new bytes(0));
```

Comment:
In case the `recipient` is a contract which doesn't implement `onERC721Received`, it won't be possible to distribute fees for the `position`. Would be good to make sure this is clearly documented somewhere if it isn't already

Reply:
good to document this definitely! in our case, if it's not the dead address, it will be a timelock contract created via the governance factory which does implement/support `onERC721Received()`.